### ISSUE TITLE
[Contribution Request] Align promptflow traces and metrics with OpenLLMetry

### ISSUE THREAD

**Issue by nirga** [microsoft/promptflow#2162]
**Is your contribution request related to a problem? Please describe.**
I'm the maintainer of [OpenLLMetry](https://github.com/traceloop/openllmetry) where we instrument LLM providers for traces and metrics on top of OpenTelemetry. 
I'd like to align the way promptflow is reporting these metrics and traces by adopting the OpenLLMetry SDK here.

**Detail the functionality and value.**
Aligning semantic conventions for LLM traces and metrics which can then be supported by different observability platforms; additional frameworks, models and vector dbs instrumented. 

**Additional context**
We're working with the [otel community](https://github.com/open-telemetry/semantic-conventions/pull/639) on making the attributes we defined the official semantic conventions, and I'm leading the [LLM semantic conventions WG](https://github.com/open-telemetry/community/blob/main/projects/llm-semconv.md) to do so.

**Comment by wangchao1230** [comment]
Thanks @nirga, promptlfow is working on integrate with open telemetry, i.e. to emit trace produced by promptflow and internall call like OpenAI with open telemetry standard. 

I wrote an [example](https://github.com/microsoft/promptflow/blob/clwan/eager-mode-sample/examples/tutorials/trace/langchain/trace_langchain.ipynb) on this, which is demostrating Tracing LangChain apps using Prompt flow & OpenTelemery. This is not merged yet, so not fully ready for customer use.
In this example, I used the use [opentelemetry-instrumentation-langchain](https://github.com/traceloop/openllmetry/tree/main/packages/opentelemetry-instrumentation-langchain) package provided by [Traceloop](https://www.traceloop.com/). Which is provided by OPenLLMetry I think. 

Is this something you epxect to align with?

**Comment by nirga** [comment]
Yes, but not just that. Also aligning on the spans created by Promptflow itself:
* Things like prompt and [token count](https://github.com/microsoft/promptflow/blob/afcb175d35b4e8ff28f87be6a5868070c12741dd/src/promptflow/promptflow/_constants.py#L114)
* Sending out the prompt key as an attribute to the OpenAI span
* Sending out workflow and task spans

**Comment by zzn2** [comment]
Thanks @nirga for reaching out to us. I have some comments in your PR, let's discuss the details there.

**Comment by nirga** [comment]
@zzn2 I haven't opened a PR yet 😅 
Wanted to see if you're onboard before I start working.

**Comment by nirga** [comment]
Thanks @zzn2! Afterwards, does it make sense to switch using these here as well?

**Comment by zzn2** [comment]
Yes. We can align with the convention when it goes stable. We can also switch some of the attributes gradually towards the new convention being reviewed.
